### PR TITLE
Create Dockerfile.{trusty,bionic,centos7}

### PR DESCRIPTION
**Pull request by Jimexist**
add a docker file for creating 0.16 thrift binary for linux (ubuntu focal)

### CODE REVIEW

**Review by trustin — APPROVED** [review]
Thanks, @Jimexist 🙇

File: lib/thrift/Dockerfile.linux [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  --without-php_extension \
+  --without-python \
+  --without-py3 \
+  --without-ruby \
+  --without-swift \
+  && \
+  make -j$(nproc) && \
+  make install
+
+FROM ubuntu:focal
```
[the remark this answers is not in the document]
Comment (trustin) [reply lib/thrift/Dockerfile.linux]: I was told by @ikhoon that their build machines are CentOS 7, which is much older than Bionic. Could you switch to Trusty as I originally suggested?

File: lib/thrift/Dockerfile.linux [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  --without-erlang \
+  --without-haskell \
+  --without-java \
+  --without-perl \
+  --without-php \
+  --without-php_extension \
+  --without-python \
+  --without-py3 \
+  --without-ruby \
+  --without-swift \
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.linux:44]:
```suggestion
  --disable-libs \
```

File: lib/thrift/Dockerfile.linux (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jimexist) [reply lib/thrift/Dockerfile.linux]: how about having 2 dockerfiles instead?

[the remark this answers is not in the document]
Comment (trustin) [reply lib/thrift/Dockerfile.linux]: Sure. That's fine as well.

**Comment by Jimexist** [comment]
@trustin fyi this is ready for review now

File: lib/thrift/Dockerfile.bionic [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# this builds a thrift binary without library support (only for code generation)
+# because Apache thrift official repo no longer maintain a binary distribution
+FROM ubuntu:bionic as builder
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# see https://archive.apache.org/dist/thrift/
+ENV THRIFT_VERSION=0.16.0
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.bionic:8]:
```suggestion
ENV THRIFT_VERSION=0.17.0
```

File: lib/thrift/Dockerfile.centos7 [hunk 4]
```diff
@@ -0,0 +1,48 @@
+
+# this builds a thrift binary without library support (only for code generation)
+# because Apache thrift official repo no longer maintain a binary distribution
+FROM centos:centos7 as builder
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# see https://archive.apache.org/dist/thrift/
+ENV THRIFT_VERSION=0.16.0
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.centos7:9]:
```suggestion
ENV THRIFT_VERSION=0.17.0
```

File: lib/thrift/Dockerfile.trusty [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# this builds a thrift binary without library support (only for code generation)
+# because Apache thrift official repo no longer maintain a binary distribution
+FROM ubuntu:trusty as builder
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# see https://archive.apache.org/dist/thrift/
+ENV THRIFT_VERSION=0.16.0
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.trusty:8]:
```suggestion
ENV THRIFT_VERSION=0.17.0
```

File: lib/thrift/Dockerfile.trusty [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  libboost-all-dev \
+  libevent-dev \
+  libssl-dev \
+  libtool \
+  make \
+  pkg-config
+
+ADD https://archive.apache.org/dist/thrift/${THRIFT_VERSION}/thrift-${THRIFT_VERSION}.tar.gz /tmp/thrift.tar.gz
+
+RUN echo "f460b5c1ca30d8918ff95ea3eb6291b3951cf518553566088f3f2be8981f6209 /tmp/thrift.tar.gz" | sha256sum -c && \
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.trusty:27]:
```suggestion
RUN echo "b272c1788bb165d99521a2599b31b97fa69e5931d099015d91ae107a0b0cc58f /tmp/thrift.tar.gz" | sha256sum -c && \
```

File: lib/thrift/Dockerfile.centos7 [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  libboost-all-dev \
+  libevent-dev \
+  libssl-dev \
+  libtool \
+  make \
+  pkg-config
+
+ADD https://archive.apache.org/dist/thrift/${THRIFT_VERSION}/thrift-${THRIFT_VERSION}.tar.gz /tmp/thrift.tar.gz
+
+RUN echo "f460b5c1ca30d8918ff95ea3eb6291b3951cf518553566088f3f2be8981f6209 /tmp/thrift.tar.gz" | sha256sum -c && \
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.centos7:29]:
```suggestion
RUN echo "b272c1788bb165d99521a2599b31b97fa69e5931d099015d91ae107a0b0cc58f /tmp/thrift.tar.gz" | sha256sum -c && \
```

File: lib/thrift/Dockerfile.bionic [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  libboost-all-dev \
+  libevent-dev \
+  libssl-dev \
+  libtool \
+  make \
+  pkg-config
+
+ADD https://archive.apache.org/dist/thrift/${THRIFT_VERSION}/thrift-${THRIFT_VERSION}.tar.gz /tmp/thrift.tar.gz
+
+RUN echo "f460b5c1ca30d8918ff95ea3eb6291b3951cf518553566088f3f2be8981f6209 /tmp/thrift.tar.gz" | sha256sum -c && \
```
Comment (ikhoon) [inline lib/thrift/Dockerfile.bionic:27]:
```suggestion
RUN echo "b272c1788bb165d99521a2599b31b97fa69e5931d099015d91ae107a0b0cc58f /tmp/thrift.tar.gz" | sha256sum -c && \
```

**Comment by ikhoon** [comment]
Sorry, I forgot this PR. Thrift 0.17.0 has been released. I updated the version and sha256 checksum for 0.17.0.

**Review by ikhoon — APPROVED** [review]
Thanks, @Jimexist! 🙇‍♂️

**Review by jrhee17 — APPROVED** [review]
Thanks @Jimexist @ikhoon ! 🙇 👍 🙇

**Review by minwoox — APPROVED** [review]
Thanks a lot, @Jimexist! 😄

**Comment by minwoox** [comment]
... and @ikhoon for taking care of this. 😊

**Pull request merged**